internal/appsec/trace/tagsholder.go

```diff
@@ -5,29 +5,58 @@
 
 package trace
 
-import "sync"
+import (
+	"encoding/json"
+	"sync"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
 
 // TagsHolder wraps a map holding tags. The purpose of this struct is to be
 // used by composition in an Operation to allow said operation to handle tags
 // addition/retrieval.
 type TagsHolder struct {
-	tags map[string]interface{}
-	mu   sync.Mutex
+	tags             map[string]any
+	tagsMu           sync.RWMutex
+	serializableTags map[string]any
+	sTagsMu          sync.RWMutex
```

Comment:
I think that a better idiomatic way of doing this in Go, avoiding code duplication here and in `Tags()` is with a type-assertion on the values: you'd be able to have `tags` alone, and only check if they implement `json.Marshaler` in the for loop in `Tags`. WDYT?

Reply:
Here's what it looks like: 881648179b820ecf5ac